galsim/des/des_psfex.py

```diff
@@ -81,16 +81,16 @@ class DES_PSFEx(object):
     called an "effective PSF".  Thus, you should not convolve by the pixel profile again
     (nor integrate over the pixel).  This would effectively include the pixel twice!
 
-    @param file_name       The file name to be read in.
-    @param image_file_name The name of the fits file of the original image (needed for the
-                           WCS information in the header).  If unavailable, you may omit this
-                           (or use None), but then the returned profiles will be in image
-                           coordinates, not world coordinates.  (Default `image_file_name = None`)
-    @param wcs             Optional way to provide the WCS if you already have it loaded from the
-                           image file. (Default `wcs = None`)
-    @param dir             Optionally a directory name can be provided if the file_name does not 
-                           already include it.  (The image file is assumed to be in the same
-                           directory.) (Default `dir = None`)
+    @param file_name_or_hdu The file name to be read in, or a pyfits HDU.
```

Comment:
I don't think we want to change the name of the existing parameter.  If anyone currently uses `file_name` as a named argument, this will break their code.  It also breaks the config stuff that currently uses this parameter name (see `_req_params` below).

There are two options that I think would be ok.  First, you could add a new `hdu` parameter and have either `file_name` or `hdu` be required.  The new parameter should be put last to maintain backward compatibility for people who are relying on the current argument order.

Or second, you could leave the parameter name `file_name`, but just say that the file_name is allowed to be an HDU rather than a string, in which case that HDU is used rather than opening up the file.  This is probably the simpler option, and I think it would be fine.  Also, it would be almost identical to your existing code, just without the change to the parameter name.